### PR TITLE
Add custom failure listener support to LottieAnimationView and log network failures

### LINKED ISSUE #1302: java.net.SocketException: Network is unreachable
`api "com.airbnb.android:lottie:3.0.7"`

Nexus 5X
Android 7.1.2

```
java.lang.IllegalStateException: Unable to parse composition
	at com.airbnb.lottie.LottieAnimationView$2.onResult(LottieAnimationView.java:75)
	at com.airbnb.lottie.LottieAnimationView$2.onResult(LottieAnimationView.java:73)
	at com.airbnb.lottie.LottieTask.notifyFailureListeners(LottieTask.java:158)
	at com.airbnb.lottie.LottieTask.access$200(LottieTask.java:27)
	at com.airbnb.lottie.LottieTask$1.run(LottieTask.java:133)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6121)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
Caused by: java.net.SocketException: Network is unreachable
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:334)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:196)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:356)
	at java.net.Socket.connect(Socket.java:605)
	at com.android.okhttp.internal.Platform.connectSocket(Platform.java:113)
	at com.android.okhttp.Connection.connectSocket(Connection.java:196)
	at com.android.okhttp.Connection.connect(Connection.java:172)
	at com.android.okhttp.Connection.connectAndSetOwner(Connection.java:367)
	at com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:130)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:329)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:246)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:457)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:126)
	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.connect(DelegatingHttpsURLConnection.java:89)
	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java)
	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetworkInternal(NetworkFetcher.java:94)
	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetwork(NetworkFetcher.java:79)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:48)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:31)
	at com.airbnb.lottie.LottieCompositionFactory$1.call(LottieCompositionFactory.java:71)
	at com.airbnb.lottie.LottieCompositionFactory$1.call(LottieCompositionFactory.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
```

### PR DESCRIPTION
**Pull request by gpeal**
This changes the default failure behavior for network errors to log instead of crash.
It also gives you the ability to set and clear failure listeners to handle this yourself.
Fixes #1302

### CODE REVIEW

**Comment by gpeal** [comment]
@digitalbuddha

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/7d2a605c7dfad9c2bad189f5f87bb35d554c7c85-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/7d2a605c7dfad9c2bad189f5f87bb35d554c7c85-android28)

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/83d4997cb55faab54916331be8114cddbad24ade-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/83d4997cb55faab54916331be8114cddbad24ade-android28)

File: lottie/src/main/java/com/airbnb/lottie/utils/Utils.java [hunk 1]
```diff
@@ -242,6 +255,16 @@ public static Bitmap resizeBitmapIfNeeded(Bitmap bitmap, int width, int height)
     return resizedBitmap;
   }
 
+  /**
+   * From http://vaibhavblogs.org/2012/12/common-java-networking-exceptions/
+   */
+  public static boolean isNetworkException(Throwable e) {
```
Comment (digitalbuddha) [inline lottie/src/main/java/com/airbnb/lottie/utils/Utils.java]: 🔥

**Comment by digitalbuddha** [comment]
lgtm, tests would be nice

**Pull request merged**